src/poisson_numcodecs/poisson.py

```diff
@@ -0,0 +1,53 @@
+"""
+Numcodecs Codec implementation for Poisson noise calibration
+"""
+import numpy as np
+import numcodecs
+from numcodecs.abc import Codec
+from . import estimate
+
+### NUMCODECS Codec ###
+class Poisson(Codec):
+    """Codec for 3-dimensional Filter. The codec assumes that input data are of shape:
+    (time, x, y).
+
+    Parameters
+    ----------
+    zero_level : float
+        Signal level when no photons are recorded. 
+        This should pre-computed or measured directly on the instrument.
+    photon_sensitivity : float
+        Conversion scalor to convert the measure signal into absolute photon numbers.
+        This should pre-computed or measured directly on the instrument.
+    """
+    codec_id = "poisson"
+
+    def __init__(self, 
+                 zero_level, 
+                 photon_sensitivity, 
+                 encoded_dtype='int8', 
+                 decoded_dtype='int16',
+                 ): 
+        self.zero_level = zero_level
+        self.photon_sensitivity = photon_sensitivity
+        self.encoded_dtype = encoded_dtype
+        self.decoded_dtype = decoded_dtype
+
+    def encode(self, buf: np.array) -> bytes:
+        lookup = estimate.make_anscombe_lookup(self.photon_sensitivity)
+        encoded = estimate.lookup(buf, lookup)
```

Comment:
I think the lookup table should only be created once at the Poisson Filter object creation. The encode and decode function, as I understand them, are called every time the Zarr object is queried. I suspect it will be more efficient to just do a lookup.